azurelinuxagent/common/event.py

```diff
@@ -406,38 +410,35 @@ def add_default_parameters_to_event(event_parameters, set_values_for_agent=True)
-        :return: Event with default parameters populated (either values for agent or extension)
+        :param event: Event which parameters need to be expanded and updated.
+        :param is_agent_event: Whether the default values will be populated from the agent or not.
+        :return: List of final parameters to be appended to the event (whether agent or extension event).
         """
-        DefaultParameter = namedtuple('DefaultParameter', ['name', 'value'])
-        default_parameters = [DefaultParameter("GAVersion", CURRENT_AGENT),
-                              DefaultParameter('ContainerId', get_container_id_from_env()),
-                              DefaultParameter('OpcodeName', datetime.utcnow().__str__() if set_values_for_agent else ""),
-                              DefaultParameter('EventTid', threading.current_thread().ident if set_values_for_agent else 0),
-                              DefaultParameter('EventPid', os.getpid() if set_values_for_agent else 0),
-                              DefaultParameter("TaskName", threading.current_thread().getName() if set_values_for_agent else ""),
-                              DefaultParameter("KeywordName", '')]
-
         # Converting the event_parameters into a dictionary as it helps to easily look up and get values
-        param_names = OrderedDict([(param.name, param.value) for param in event_parameters])
+        event_params_dict = OrderedDict([(param.name, param.value) for param in event.parameters])
 
-        for param in default_parameters:
-            if param.name not in param_names or set_values_for_agent:
-                # If set_values_for_agent, we disregard any values already set for an existing default property and
-                # replaces it with a latest entry.
-                param_names[param.name] = param.value
-
-        parameters = DataContractList(TelemetryEventParam)
-        for name, value in param_names.items():
-            parameters.append(TelemetryEventParam(name, value))
-
-        return parameters
+        DefaultParameter = namedtuple('DefaultParameter', ['name', 'value'])
+        default_params = [DefaultParameter('GAVersion', CURRENT_AGENT),
+                          DefaultParameter('ContainerId', get_container_id_from_env()),
+                          DefaultParameter('OpcodeName', datetime.utcnow().__str__() if is_agent_event else event_params_dict.get('OpcodeName', '')),
+                          DefaultParameter('EventTid', threading.current_thread().ident if is_agent_event else event_params_dict.get('EventTid', 0)),
+                          DefaultParameter('EventPid', os.getpid() if is_agent_event else event_params_dict.get('EventPid', 0)),
+                          DefaultParameter('TaskName', threading.current_thread().getName() if is_agent_event else event_params_dict.get('TaskName', '')),
+                          DefaultParameter('KeywordName', event_params_dict.get('KeywordName', ''))]
```

Comment:
For extension events, I decided to let them keep the existing value of these parameters, otherwise set them to empty values. I did this because I'm not sure if any extension publishers might rely on these fields for some sort of debugging (like we are using `OpcodeName` to tell us the event-creation time for agent events, for example).

Let me know if you think we should always override them. Currently we only always override `GAVersion` and `ContainerId`.

Reply:
No I think we should let them keep what they populate (infact that is what we're advertising they do in the new telemetry pipeline too).

Reply:
This is outdated; the current design only keeps extension-specific fields for extension events. Refer to the description of this PR for more details.